### DISCUSSION TITLE
*.gradle.kts should be a data file

### DISCUSSION THREAD

**Discussion by anatawa12** [github-linguist/linguist#5352] [General]
For the `.gradle` extension, the file is not listed on the Languages list for a repository. For example, https://github.com/anatawa12/ForgeGradle-example or https://github.com/anatawa12/ForgeGradle-2.3.

I think it's because the Gradle file is a data file.

https://github.com/github/linguist/blob/32ec19c013a7f81ffaeead25e6e8f9668c7ed574/lib/linguist/languages.yml#L2046-L2052

However, Because of no extra definition for`.gradle.kts` extension, `.gradle.kts` will be listed as a Kotlin file in the Languages list for a repository. For example, https://github.com/anatawa12/ForgeGradle-2.3 ([list of kt/kts/ktm file](https://github.com/anatawa12/ForgeGradle-2.3/search?q=extension%3Akt+extension%3Akts+extension%3Aktm)).

I think It's not good. I think If `*.gradle` is not listed, `*.gradle.kts` should not be listed, and if `*.gradle` is listed, `*.gradle.kts` should be listed.

**Comment by mikewacker** [comment]
Yeah, `.gradle` does not get listed as Groovy, even though it uses the Groovy DSL. Likewise, `.gradle.kts` should not get listed as Kotlin, even though it uses the Kotlin DSL. As a general principle, build code not not be treated as actual code.

**Comment by lildude** [comment]
This should be working now https://github.com/github-linguist/linguist/pull/6541 has been merged and deployed.
